src/main/java/com/google/error_analyzer/backend/FulltextSearchQuery.java

```diff
@@ -0,0 +1,131 @@
+            ArrayList<ErrorLine> errorData = getErrors(indexFile,client);
+            Gson gson = new Gson();
+            String json = gson.toJson(errorData);
+            return json;
+        }catch(IOException e){
+            logger.error("Could not complete query request: "+e);
+            return "Could not complete request.";
+        }
+
+    }
+
+    //returns all search hits as Errorline object array
+    public ArrayList<ErrorLine> getErrors(String indexFile, RestHighLevelClient client) throws IOException{
+        ArrayList<ErrorLine> errorData = new ArrayList<>();
+        SearchHits hits = getQueryHits(indexFile,client);
+        errorData = getLogData(hits);
+        return errorData;
+    }
+
+    //Make the search request and  return Search hits from match query.
+    private SearchHits getQueryHits(String indexFile, RestHighLevelClient client) throws IOException{
+        SearchRequest searchRequest = new SearchRequest(indexFile);
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder(); 
+
+        // kewords class contains all the terms used combined in OR logic.
+        Keywords errorKeywords = new Keywords();
+        searchSourceBuilder.query(
+            QueryBuilders.matchQuery(
+                logTextField, errorKeywords.getQueryString() )); 
+        searchRequest.source(searchSourceBuilder); 
+
+        SearchResponse searchResponse = client.search(searchRequest, RequestOptions.DEFAULT);
+        SearchHits hits = searchResponse.getHits();
+        return hits;
+    }
+
+    // return log text from a hit.
+    private String extractLogText(SearchHit hit){
+        String sourceAsString = hit.getSourceAsString();
+        Map<String, Object> sourceAsMap = hit.getSourceAsMap();
+        return (String) sourceAsMap.get(logTextField);
+    }
+
+    // return log line number from a hit 
+    private int extractLogLineNumber(SearchHit hit){
+        String sourceAsString = hit.getSourceAsString();
+        Map<String, Object> sourceAsMap = hit.getSourceAsMap();
+        //subject to the format logLineNumber is stored in index, string or integer.
+        try{
+            int lineNumber = Integer.parseInt((String) sourceAsMap.get(logLineNumberField));
+            logger.info("Integer.parseInt successfully = " + lineNumber);
+            return lineNumber;
+        }catch(NumberFormatException e){
+            logger.error("Integer.parseInt error: " + e);
+            return -1;
+        }catch(NullPointerException e2){
+            logger.error("Null error: " + e2);
+            return -1;
+        }
+    }
+    
+    //Create Errorline array from search hits.
+    private ArrayList<ErrorLine> getLogData(SearchHits hits){
+
+        ArrayList<ErrorLine> errorMsgs = new ArrayList<>();
+        for (SearchHit hit : hits){
+            int lineNumber = extractLogLineNumber(hit);
+            if( lineNumber != -1){
+                errorMsgs.add(new ErrorLine( extractLogText(hit),lineNumber) );
```

Comment:
Will there be a case when line numbers might repeat?

Reply:
We should face this issue. While storing logs, logLineNumber is also the id of the document